### DISCUSSION TITLE
How to fix "Module '"$env/static/public"' has no exported member 'PUBLIC_SITE_NAME'" error

### DISCUSSION THREAD

**Discussion by tauflife** [scosman/CMSaasStarter#43] [Q&A] [answered]
Hi,

Full disclaimer, I'm a sveltekit noob, so this may be a total beginner question. I typically create mobile and web apps with Flutter and Dart, but am looking to get into using sveltekit and this seems like it's a great project to learn from.

I believe I've followed the quickstart directions correctly (through the supabase instructions). Haven't implemented Stripe stuff yet, but I think the project should build ok without it (maybe not)?

Anyways, when I pull up the site on localhost I get about 1 second of the site that I expect, then a `"this is embarrassing, internal error"` message.

The errors I'm getting is:

```
src/routes/(marketing)/+page.svelte:164:12
Error: Module '"$env/static/public"' has no exported member 'PUBLIC_SITE_NAME'. (ts)

src/routes/(marketing)/pricing/+page.svelte:3:12
Error: Module '"$env/static/public"' has no exported member 'PUBLIC_SITE_NAME'. (ts)
```

I also get the Stripe error messages, but that makes sense to me b/c I haven't implemented that yet. Just for reference, here's the errors I get about that:

```
src/routes/(admin)/account/subscription_helpers.server.ts:5:10
Error: Module '"$env/static/private"' has no exported member 'PRIVATE_STRIPE_API_KEY'. 
import { pricingPlans } from "../../(marketing)/pricing/pricing_plans"
import { PRIVATE_STRIPE_API_KEY } from "$env/static/private"
import Stripe from "stripe"

src/routes/(admin)/account/(menu)/billing/manage/+page.server.ts:4:10
Error: Module '"$env/static/private"' has no exported member 'PRIVATE_STRIPE_API_KEY'. 
import type { PageServerLoad } from "./$types"
import { PRIVATE_STRIPE_API_KEY } from "$env/static/private"
import Stripe from "stripe"

src/routes/(admin)/account/subscribe/[slug]/+page.server.ts:7:10
Error: Module '"$env/static/private"' has no exported member 'PRIVATE_STRIPE_API_KEY'. 
import type { PageServerLoad } from "./$types"
import { PRIVATE_STRIPE_API_KEY } from "$env/static/private"
import Stripe from "stripe"
```

Here's the versions I'm working with:

```
├── @supabase/auth-helpers-sveltekit@0.10.7
├── @supabase/auth-ui-svelte@0.2.9
├── @sveltejs/adapter-auto@2.1.1
├── @sveltejs/kit@1.30.4
├── eslint-plugin-svelte@2.35.1
├── prettier-plugin-svelte@3.2.2
├── svelte-check@3.6.7
├── svelte@4.2.12
```

From what I've looked into, this is a typescript issue (sorry if that's obvious, remember, I'm a noob). I think I might just not have generated my required types, but I'm not sure.

I was reading [this](https://github.com/sveltejs/kit/issues/8882).

Anyways, if anyone can give any info on how to resolve this that would be great.

I feel like the instructions have been great for this project. I just don't know how to solve this problem.

Thanks!

**Comment by scosman** [comment, marked as answer]
You just need to setup the right environment variables. 

I made the setup instructions a bit clearer for this yesterday, but you may have forked before that. 

Get the latest and run 'cp local_env_template .env.local'. It will create dummy values that makes the app functional.

**Reply by tauflife** [reply]
Thanks a lot @scosman 
I actually went and read that in the online readme, but didn't see the `local_env_template` in my forked project. Thought it was a mistake. Guess not! Thanks!
